### PR TITLE
Update validator guide with restricted sync committee computation

### PR DESCRIPTION
**Pull request by ralexstokes**
Depends on #2406.

Some of the language in the validator guide no longer makes much sense in light of the changes in #2406.

This PR suggests an alternative.

### CODE REVIEW

File: specs/altair/validator.md [hunk 1]
```diff
@@ -177,7 +177,6 @@ At any given `epoch`, the `BeaconState` contains the current `SyncCommittee` and
 Once every `EPOCHS_PER_SYNC_COMMITTEE_PERIOD` epochs, the next `SyncCommittee` becomes the current `SyncCommittee` and the next committee is computed and stored.
 
 *Note*: The data required to compute a given committee is not cached in the `BeaconState` after committees are calculated at the period boundaries.
-This means that calling `get_sync_commitee()` in a given `epoch` can return a different result than what was computed during the relevant epoch transition.
```
Comment (ralexstokes) [inline specs/altair/validator.md:180]: at this point, `get_{next_,}sync_committee` is an internal function to the state transition

File: specs/altair/validator.md [hunk 2]
```diff
@@ -229,12 +228,12 @@ def process_sync_committee_contributions(block: BeaconBlock,
...
     sync_aggregate = SyncAggregate()
     signatures = []
+    sync_subcommittee_size = SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT
 
     for contribution in contributions:
         subcommittee_index = contribution.subcommittee_index
         for index, participated in enumerate(contribution.aggregation_bits):
             if participated:
-                sync_subcommittee_size = SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT
                 participant_index = sync_subcommittee_size * subcommittee_index + index
```
Comment (ralexstokes) [inline specs/altair/validator.md:237]: non-substantive refactor bc i had already opened this file back up

File: specs/altair/validator.md [hunk 3]
```diff
@@ -367,7 +366,7 @@ Set `contribution.subcommittee_index` to the index for the subcommittee index co
 ###### Aggregation bits
 
 Let `contribution.aggregation_bits` be a `Bitvector[SYNC_COMMITTEE_SIZE // SYNC_COMMITTEE_SUBNET_COUNT]`, where the `index`th bit is set in the `Bitvector` for each corresponding validator included in this aggregate from the corresponding subcommittee.
-An aggregator finds the index in the sync committee (as returned by `get_sync_committee_indices()`) for a given validator referenced by `sync_committee_signature.validator_index` and maps the sync committee index to an index in the subcommittee (along with the prior `subcommittee_index`). This index within the subcommittee is set in `contribution.aggegration_bits`.
+An aggregator finds the index in the sync committee (as determined by a reverse pubkey lookup on `state.current_sync_committee.pubkeys`) for a given validator referenced by `sync_committee_signature.validator_index` and maps the sync committee index to an index in the subcommittee (along with the prior `subcommittee_index`). This index within the subcommittee is set in `contribution.aggegration_bits`.
```
Comment (ralexstokes) [inline specs/altair/validator.md:369]: will readers know what a "reverse pubkey lookup" is?

for reference, there is one in `compute_subnets_for_sync_committee`

Comment (djrtwo) [reply specs/altair/validator.md:369]: likely!

**Review by djrtwo — APPROVED** [review]

**Pull request merged**